examples/perplexity/perplexity.cpp

```diff
@@ -368,7 +368,7 @@ results_perplexity perplexity(llama_context * ctx, const gpt_params & params) {
         // Example, we have a context window of 512, we will compute perplexity for each of the
         // last 256 tokens.  Then, we split the input up into context window size chunks to
         // process the entire prompt.
-        const int first = std::min(512, params.n_ctx/2);
+        const int first = params.n_ctx/2;
```

Comment:
Needs some extra eyes on this

Reply:
This was introduced in commit 864dcb26fb2f74453f223a9cc100768530e7a391 by @glinscott. I think the idea is that the model shouldn't need much more than 512 tokens in order to predict the rest of the context.

This is kind of inverted from the idea of strided perplexity in the [reference implementation](https://huggingface.co/docs/transformers/perplexity) - where I believe it computes the likelihood of the _last_ 512 tokens, based on the first n_ctx - 512.

Reply:
Ah, this std::min was added to ensure that the model has a reasonable amount of context to do the prediction based off.  The huggingface implementation assumes a fixed context window size, while folks are able to use different context window lengths here.  If it's too small, the perplexity goes way up.  But, probably easier to do that as a restriction on the context window size, or a warning if it's too small. 

This came from https://github.com/ggerganov/llama.cpp/pull/407/files, where I also set the default batch size to 512 as well.

Reply:
The batch size at some point was restricted to max of 512 due to memory limitations. Today, it should be fine to run with larger batches.

Regarding this line, I think it's fine to remove the `std::min`.
Strided perplexity is correctly implemented in `perplexit_v2` above.